tests/dialects/test_postgres.py

```diff
@@ -1498,3 +1498,78 @@ def test_locks(self):
         for key_type in ("FOR SHARE", "FOR UPDATE", "FOR NO KEY UPDATE", "FOR KEY SHARE"):
             with self.subTest(f"Test lock type {key_type}"):
                 self.validate_identity(f"SELECT 1 FROM foo AS x {key_type} OF x")
+
+    def test_grant(self):
```

Comment:
These were not in the original [PR](https://github.com/tobymao/sqlglot/pull/4138), so I added them.